### PR TITLE
feat(ui): Add the ability to have links to logging facility in UI. Closes #2438

### LINKED ISSUE #2438: Deep link into Splunk (or other logging facility) from UI
Provide a way to get directly from the UI to the logs for the pod.

See #2321

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [x] Optional. I've added My organization is added to the USERS.md.
* [x] I've signed the CLA and required builds are green. 

Closes #2438

Screen Shot 2020-03-16 at 5 46 51 PM
Screen Shot 2020-03-16 at 5 46 53 PM

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/logging-facility.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package v1alpha1
+
+type LoggingFacility struct {
+	Name      string    `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+	Templates Templates `json:"templates,omitempty" protobuf:"bytes,2,opt,name=templates"`
+}
+
+type Templates struct {
```
Comment (alexec) [inline pkg/apis/workflow/v1alpha1/logging-facility.go]: rename to `LoggingFacilityTemplates`

File: ui/src/app/workflows/components/workflow-details/workflow-details.tsx [hunk 2]
```diff
@@ -366,4 +384,8 @@ export class WorkflowDetails extends React.Component<RouteComponentProps<any>, {
     private get appContext(): AppContext {
         return this.context as AppContext;
     }
+
+    private openLink(link: Link) {
```
Comment (whynowy) [inline ui/src/app/workflows/components/workflow-details/workflow-details.tsx]: This is great, just a minor suggestion, I think a new window/tab would be better.

Comment (alexec) [reply ui/src/app/workflows/components/workflow-details/workflow-details.tsx]: good idea, I prefer ctrl+click to open a new window - I'll check that works

Comment (alexec) [reply ui/src/app/workflows/components/workflow-details/workflow-details.tsx]: ok, we can't do ctrl+click :(

**Comment by salanki** [comment]
This feature is gold. Thank you!

**Pull request merged**